test/tests/functional/pbs_snapshot_unittest.py

```diff
@@ -94,12 +95,22 @@ def setup_sc(self, sched_id, partition, port):
         :param log_filter: log filter value for the scheduler
         :type log_filter: int
         """
-        a = {'partition': partition,
-             'sched_host': self.server.hostname,
-             'sched_port': port}
+        if sched_priv is not None:
```

Comment:
What if sched_log is None? you are not checking for that. Suggestion: create attribute list with just 'partition', 'sched_host' and 'sched_port'. Then, add 'sched_priv' or 'sched_log' if they are not None.